src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -52,6 +52,15 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Queryable Entities
 
+(defn augment-for-subquery
+  [{:keys [entity] :as query-rec}]
+  (case entity
```

Comment:
This is equivalent to `(cond-> query-rec (= :facts entity) (assoc-in [:projections ....))`, which I personally like a little more because it calls out what's being transformed/returned over the conditional